### PR TITLE
Fixes 10343: Add methods to update Glossary and GlossaryTerm in Python SDK

### LINKED ISSUE #10343: Add method to update GlossaryTerm in Python SDK
**Is your feature request related to a problem? Please describe.**
When creating Glossaries from SDK, I'm using `CreateGlossaryTermRequest`  with `create_glossary_term` method, but it can only create new term and it will not update existing terms.

**For example:**
We have this as a first pass:
`Parent A > Child 1`
`Parent B > Child 2`
Then on second pass we want to change Child 2's parent to Parent A, then we get:
`Parent A > Child 1`
`Parent A > Child 2`
`Parent B > Child 2`
`Child 2` is created as a new term, and not updated the existing term.

**Describe the solution you'd like**
We could have a simple method `create_or_update_glosary_term` using the API endpoint [PUT /v1/glossaryTerms](https://docs.open-metadata.org/swagger.html#operation/createOrUpdateGlossaryTerm)
 
As discussed on Slack: https://openmetadata.slack.com/archives/C02B6955S4S/p1677233629287539?thread_ts=1677228836.335459&cid=C02B6955S4S

**Describe alternatives you've considered**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by schlameel**
### Describe your changes:

Fixes #10343  

- added methods to [mixins/glossary_mixin.py](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py) to PATCH:
  - GlossaryTerm parent
  - GlossaryTerm related terms
  - GlossaryTerm synonyms
  - GlossaryTerm references
  - reviewers for Glossary and GlossaryTerm
- created a super class, OMetaPatchMixinBase, in mixins/patch_mixin_utils.py for mixins that PATCH content. Multiple mixins use a common method. Updated mixins to inherit the new super class.
- moved common PATCH enums from ometa/patch.py to ometa/mixins/patch_mixin_utils.py and updated imports where used.
- created tests for all new methods as well as for some basic, pre-existing methods as they relate to Glossary and GlossaryTerm (create_or_update).

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

Improvement
- [x] I have added tests around the new logic.
- [ ] For connector/ingestion changes: I updated the documentation.

New feature
- [x] The issue properly describes why the new feature is needed, what's the goal, and how we are building it. Any discussion
    or decision-making process is reflected in the issue.
- [ ] I have updated the documentation.
- [x] I have added tests around the new logic.

### CODE REVIEW

**Review by harshach — APPROVED** [review]

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py [hunk 1]
```diff
@@ -13,19 +13,32 @@
...
 from pydantic import BaseModel
 
 from metadata.generated.schema.entity.data.glossary import Glossary
 from metadata.generated.schema.entity.data.glossaryTerm import GlossaryTerm
+from metadata.generated.schema.type import basic
+from metadata.ingestion.ometa.utils import model_str
 from metadata.utils.logger import ometa_logger
 
-T = TypeVar("T", bound=BaseModel)
+from .patch_mixin_utils import (
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py:28]: can we have imports with the full path here?

**Review by pmbrull — COMMENTED** [review]
Thanks for this. Just left a nit comment to align styling. LGTM

File: ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py (hunk 1 above)
Comment (schlameel) [reply ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py]: Fixed

**Review by pmbrull — APPROVED** [review]

**Comment by pmbrull** [comment]
LGTM, looks like you might be missing a `make py_format` though

```
isort --check-only ingestion/ openmetadata-airflow-apis/ --skip ingestion/src/metadata/generated --skip ingestion/build --profile black --multi-line 3
ERROR: /home/runner/work/OpenMetadata/OpenMetadata/ingestion/src/metadata/ingestion/ometa/mixins/glossary_mixin.py Imports are incorrectly sorted and/or formatted.
Skipped 2 files
```

**Pull request merged**